### PR TITLE
feat(hits): Add BEM styling to the `hits` widget

### PR DESCRIPTION
**Pull request by pixelastic**
- Add BEM classes to the hit list and each hit element
- Add one when the list is empty
- Renamed the key from `hit` to `item` in CSS class, template and
  transformData to stay consistent with other widgets
- Updated the doc accordingly, with styling examples
- Updated tests
- Updated the bemHelper to be able to use a modifier without an
  element (`ais-hits__empty`)

BREAKING CHANGE: The hit template and transform data key is renamed
from `hit` to `item` to stay consistent with the other widgets

### CODE REVIEW

File: components/Hits.js [hunk 1]
```diff
@@ -5,23 +5,26 @@ var Template = require('./Template');
...
+        </div>
       );
     });
 
-    return <div>{renderedHits}</div>;
+    return <div className={this.props.cssClasses.root}>{renderedHits}</div>;
   }
 
   renderNoResults() {
+    var className = [this.props.cssClasses.root, this.props.cssClasses.empty].join(' ');
```
Comment (bobylito) [inline components/Hits.js]: Why not just : `this.props.cssClasses.root + ' ' + this.props.cssClasses.empty` ?

**Comment by bobylito** [comment]
One comment, otherwise looks great :)

File: components/Hits.js (hunk 1 above)
Comment (vvo) [inline components/Hits.js]: template strings or classname module which can do this. indeed

Comment (pixelastic) [inline components/Hits.js]: Yep, right. Sometimes the obvious is not that obvious :)

Comment (bobylito) [inline components/Hits.js]: :D

Comment (redox) [inline components/Hits.js]: and I would go for `cx(this.props.cssClasses.root, this.props.cssClasses.empty)` :)

Comment (bobylito) [inline components/Hits.js]: Isn't that like overkill :D (but consistent with the rest okkkk :))

Comment (pixelastic) [inline components/Hits.js]: Won't it be overkill ? requiring a library, that will surely do an array + join and dedupe just to concatenate two strings?

Comment (bobylito) [inline components/Hits.js]: Jinx!

**Pull request merged**